tachyon/zk/plonk/vanishing/vanishing_utils.h

```diff
@@ -0,0 +1,166 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_VANISHING_VANISHING_UTILS_H_
+#define TACHYON_ZK_PLONK_VANISHING_VANISHING_UTILS_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/parallelize.h"
+
+namespace tachyon::zk {
+
+// Calculate ζ = g^((2ˢ * T) / 3).
+template <typename F>
+constexpr F GetZeta() {
+  CHECK_EQ(F::Config::kTrace % math::BigInt<F::kLimbNums>(3),
+           math::BigInt<F::kLimbNums>(0));
+  return F::FromMontgomery(F::Config::kSubgroupGenerator)
+      .Pow(F(2).Pow(F::Config::kTwoAdicity).ToBigInt() * F::Config::kTrace /
+           math::BigInt<F::kLimbNums>(3));
+}
+
+// Calculate ζ⁻¹ = g^((2ˢ * T) / 3).
+// NOTE(TomTaehoonKim): Halo2 calculates ζ = g^((2ˢ * T) / 3)². So in need of
+// ζ, square the result of this function.
+template <typename F>
+constexpr F GetHalo2ZetaInv() {
+  return GetZeta<F>();
+}
```

Comment:
What I meant at office is different from this. So you can 

```suggestion
// NOTE(TomTaehoonKim): The returning value of |GetZeta()| is different from the one in
// https://github.com/kroma-network/halo2curves/blob/c0ac1935e5da2a620204b5b011be2c924b1e0155/src/bn256/fr.rs#L112-L118.
// This is an ugly way to produce a same result with Halo2Curves but we will
// remove once we don't have to match it against Halo2 any longer in the
// future.
// Calculate ζ = g^((2ˢ * T) / 3)².
template <typename F>
constexpr F GetHalo2Zeta() {
  return GetZeta<F>().Sqaure();
}
```